changes/2020-06-17_encrypt-max-plaintext-length-input/change.md

```diff
@@ -0,0 +1,179 @@
+[//]: # "Copyright Amazon.com Inc. or its affiliates. All Rights Reserved."
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+# Encrypt Max Plaintext Length Input
+
+## Affected Features
+
+| Feature                                           |
+| ------------------------------------------------- |
+| [Encrypt](../../client-apis/encrypt.md)           |
+| [CMM Interface](../../framework/cmm-interface.md) |
+
+## Affected Specifications
+
+| Specification                                     |
+| ------------------------------------------------- |
+| [Encrypt](../../client-apis/encrypt.md)           |
+| [CMM Interface](../../framework/cmm-interface.md) |
+
+## Affected Implementations
+
+| Language   | Repository                                                                            |
+| ---------- | ------------------------------------------------------------------------------------- |
+| Python     | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-python)              |
+| Javascript | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-javascript) |
+
+## Definitions
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+### Known-Length Plaintext
+
+Any plaintext input to the Encrypt where the total length of that plaintext
+can be immediately determined.
+For example, if a user supplies plaintext as a string or byte array to Encrypt,
+that is a known-length plaintext.
+
+### Unknown-Length Plaintext
+
+Any plaintext input to the Encrypt operation where the total length of that plaintext
+cannot be immediately determined.
+For example, if a user supplies plaintext as a stream to Encrypt,
+that is a unknown-length plaintext.
+
+## Summary
+
+When performing the encrypt operation on an unknown length plaintext,
+users MUST be able to specify some input that bounds the length of the plaintext to be encrypted.
```

Comment:
Should the summary also state that users MUST NOT be able to specify the maximum plaintext length when performing Encrypt on a known-length plaintext?

Reply:
We need to be careful about the wording, but yes. I've updated the summary.